kodein-android/src/main/java/com/github/salomonbrys/kodein/android/KodeinAndroidComponents.kt

```diff
@@ -0,0 +1,363 @@
+package com.github.salomonbrys.kodein.android
+
+import android.app.*
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.support.v4.app.FragmentActivity
+import android.support.v7.app.AppCompatActivity
+import android.view.LayoutInflater
+import com.github.salomonbrys.kodein.*
+import android.support.v4.app.Fragment as SupportFragment
+import android.support.v4.app.FragmentManager as SupportFragmentManager
+import android.support.v4.app.LoaderManager as SupportLoaderManager
+
+private fun inject(injector: KodeinInjected, componentModule: Kodein.Module, superKodein: Kodein) {
+  val kodein = Kodein {
+    extend(superKodein, allowOverride = true)
+
+    import(componentModule, allowOverride = true)
+  }
+
+  injector.inject(kodein)
+}
+
+interface AndroidInjector<T, out S : AndroidScope<T>> : KodeinInjected {
+  // this covers most of the use cases (and it really shouldn't ever be in a position where the cast doesn't work)
+  @Suppress("UNCHECKED_CAST")
+  val component: T get() = this as T
+  val scope: S
+
+  fun initializeInjector()
+
+  fun destroyInjector() = scope.removeFromScope(component)
+
+  fun provideOverridingBindings(): Kodein.Builder.() -> Unit = {}
+}
+
+interface ActivityInjector : AndroidInjector<Activity, AndroidScope<Activity>> {
+  override val scope: AndroidScope<Activity> get() = androidActivityScope
+
+  override fun initializeInjector() {
+    val activityModule = Kodein.Module {
+      bindGeneric<KodeinInjected>() with erasedInstance(this@ActivityInjector)
```

Comment:
Nope.
You really need `bindGeneric` if you are binding a generic type (such as `List<String>`).

All those types (`KodeinInjected`, `Context`, `Activity`, `FragmentManager`, `LoaderManager` and `LayoutInflater`) are *not* generic.

Therefore, you should use `bindErased`. It will do exactly the same as `bindGeneric`, but with better perf.